presto-main/src/main/java/io/prestosql/sql/planner/iterative/IterativeOptimizer.java

```diff
@@ -57,17 +58,19 @@
     private final CostCalculator costCalculator;
     private final List<PlanOptimizer> legacyRules;
     private final RuleIndex ruleIndex;
+    private final Predicate<Session> useLegacyRules;
 
     public IterativeOptimizer(RuleStatsRecorder stats, StatsCalculator statsCalculator, CostCalculator costCalculator, Set<Rule<?>> rules)
     {
-        this(stats, statsCalculator, costCalculator, ImmutableList.of(), rules);
+        this(stats, statsCalculator, costCalculator, session -> false, ImmutableList.of(), rules);
     }
 
-    public IterativeOptimizer(RuleStatsRecorder stats, StatsCalculator statsCalculator, CostCalculator costCalculator, List<PlanOptimizer> legacyRules, Set<Rule<?>> newRules)
+    public IterativeOptimizer(RuleStatsRecorder stats, StatsCalculator statsCalculator, CostCalculator costCalculator, Predicate<Session> useLegacyRules, List<PlanOptimizer> legacyRules, Set<Rule<?>> newRules)
```

Comment:
Is there a way to set this useLegacy session rules and pass the legacy rule ?

Reply:
No directly. This is callsite specific. For instance, you might create a new set of rules with a new session property to enable/disable the legacy rule just for that optimization, and in the instantiation of the IterativeOptimizer you'd pass a predicate that knows how to check whether that session is set.